examples/sequence_tagging/conll_reader.py

```diff
@@ -0,0 +1,266 @@
+    """
+    glove_dict = dict()
+    with open(filename) as fin:
+        for line in fin:
+            vec = line.strip().split()
+            if len(vec) == 0:
+                continue
+            word, vec = vec[0], vec[1:]
+            word = tx.utils.compat_as_text(word)
+            word = DIGIT_RE.sub("0", word) if normalize_digits else word
+            glove_dict[word] = np.array([float(v) for v in vec])
+            if len(vec) != emb_dim:
+                raise ValueError("Inconsistent word vector sizes: %d vs %d" %
+                                 (len(vec), emb_dim))
+    return glove_dict
+
+
+def construct_init_word_vecs(vocab, word_vecs, glove_dict):
+    for word, index in vocab.items():
+        if word in glove_dict:
+            embedding = glove_dict[word]
+        elif word.lower() in glove_dict:
+            embedding = glove_dict[word.lower()]
+        else:
+            embedding = None
+
+        if embedding is not None:
+            word_vecs[index] = embedding
+    return word_vecs
+
+
+class CoNLLReader:
+
+    def __init__(self, file_path, word_vocab, char_vocab, ner_vocab):
+        self.__source_file = open(file_path, 'r', encoding='utf-8')
+        self.__word_vocab = word_vocab
+        self.__char_vocab = char_vocab
+        self.__ner_vocab = ner_vocab
+
+    def close(self):
+        self.__source_file.close()
+
+    def getNext(self, normalize_digits=True):
+        line = self.__source_file.readline()
+        # skip multiple blank lines.
+        while len(line) > 0 and len(line.strip()) == 0:
+            line = self.__source_file.readline()
+        if len(line) == 0:
+            return None
+
+        lines = []
+        while len(line.strip()) > 0:
+            line = line.strip()
+            lines.append(line.split(' '))
+            line = self.__source_file.readline()
+
+        length = len(lines)
+        if length == 0:
+            return None
+
+        words = []
+        word_ids = []
+        char_seqs = []
+        char_id_seqs = []
+        ner_tags = []
+        ner_ids = []
+
+        for tokens in lines:
+            chars = []
+            char_ids = []
+            for char in tokens[1]:
+                chars.append(char)
+                char_ids.append(self.__char_vocab[char])
+            if len(chars) > MAX_CHAR_LENGTH:
+                chars = chars[:MAX_CHAR_LENGTH]
+                char_ids = char_ids[:MAX_CHAR_LENGTH]
+            char_seqs.append(chars)
+            char_id_seqs.append(char_ids)
+
+            word = DIGIT_RE.sub("0", tokens[1]) \
+                if normalize_digits else tokens[1]
+            ner = tokens[4]
+
+            words.append(word)
+            word_ids.append(self.__word_vocab[word])
+
+            ner_tags.append(ner)
+            ner_ids.append(self.__ner_vocab[ner])
+
+        return NERInstance(Sentence(words, word_ids, char_seqs, char_id_seqs),
+                           ner_tags, ner_ids)
+
+
+class NERInstance:
```

Comment:
This and `Sentence` below can be changed to `NamedTuple`s, which also supports custom methods.